app/helpers/ExerciseConfig/Compiler.php

```diff
@@ -18,6 +18,8 @@
  */
 class Compiler {
 
+  public const EXERCISE_CONFIG_TYPES = ["simpleExerciseConfig", "advancedExerciseConfig"];
```

Comment:
Whye here? It is not related to compilation, only to displaying data in web-app, I'd rather see this "list" in entity just like default parameters.

Reply:
Well, I thought about the entity too, but I don't think it belongs there either. There might be more compilation types in the future serviced by different compilers. Feel free to suggest a better place.

Reply:
Well, it is true that storing new configuration into ExerciseConfig entity might be one of the solutions. But it somehow smells. If you want to do this properly... then in the Compiler do a switch/if over config types...

Reply:
Okay, I'll do that.

Reply:
This of course means that you have to add type into Assignment entity...

Reply:
Why? I can access the exercise through the assignment entity, right?

Reply:
Yes, but what if type of the exercise will be changed in the meantime? Supervisor will use different type of confiugration, because he/she want to do exercise a bit more difficult...

Reply:
Oh yeah, Assignments can override ExerciseConfig... I need to think about this for a bit.

Reply:
well... cannot for now and I think we'd like to keep it that way. But in the database by manual override... yes...

Reply:
OK, copyiing config type to Assignment is probably the best option. Yuck.